### PR TITLE
[New Rule] Potential Credential Access via DCSync

### PR DESCRIPTION
**Pull request by w0rk3r**
## Issues

Related to #1490 

## Summary

This rule identifies when a User Account starts the Active Directory Replication Process. Attackers can use the DCSync technique to get credential information of individual accounts or the entire domain, thus compromising the entire domain.

<details>
<summary>Example Event 4662</summary>

```
{
  "_index": ".ds-logs-system.security-default-2022.02.02-000002",
  "_type": "_doc",
  "_id": "Uxjc3n4BWi-ekcTSaL7k",
  "_score": 1,
  "_source": {
    "agent": {
      "hostname": "DCK",
      "name": "DCK",
      "id": "e9c742aa-7cc9-4ec9-b8ca-fe3e493552a8",
      "ephemeral_id": "1101fb45-3ad8-4ec4-9fbc-2e40a4801624",
      "type": "filebeat",
      "version": "7.16.3"
    },
    "winlog": {
      "computer_name": "DCK.kerb.com",
      "process": {
        "pid": 512,
        "thread": {
          "id": 13580
        }
      },
      "keywords": [
        "Audit Success"
      ],
      "logon": {
        "id": "0x6327a21"
      },
      "channel": "Security",
      "event_data": {
        "AdditionalInfo": "-",
        "SubjectLogonId": "0x6327a21",
        "Properties": "%%7688\n\t\t{1131f6ad-9c07-11d1-f79f-00c04fc2dcd2}\n\t{19195a5b-6da0-11d0-afd3-00c04fd930c9}",
        "AccessMask": "0x100",
        "ObjectName": "%{da8c583e-6d51-41d5-8515-782cba5f390b}",
        "ObjectType": "%{19195a5b-6da0-11d0-afd3-00c04fd930c9}",
        "SubjectUserName": "Administrator",
        "AccessListDescription": [
          "Control Access"
        ],
        "ObjectServer": "DS",
        "SubjectDomainName": "KERB",
        "HandleId": "0x0",
        "OperationType": "Object Access",
        "AccessMaskDescription": [
          "Control Access"
        ],
        "AccessList": "%%7688\n\t\t\t\t",
        "SubjectUserSid": "S-1-5-21-2581715867-511872244-182302923-500"
      },
      "opcode": "Info",
      "record_id": "12593",
      "event_id": "4662",
      "task": "Directory Service Access",
      "provider_guid": "{54849625-5478-4994-A5BA-3E3B0328C30D}",
      "activity_id": "{E2864E26-1C39-0000-4F4E-86E2391CD801}",
      "api": "wineventlog",
      "provider_name": "Microsoft-Windows-Security-Auditing"
    },
    "log": {
      "level": "information"
    },
    "elastic_agent": {
      "id": "e9c742aa-7cc9-4ec9-b8ca-fe3e493552a8",
      "version": "7.16.3",
      "snapshot": false
    },
    "message": "An operation was performed on an object.\n\nSubject :\n\tSecurity ID:\t\tS-1-5-21-2581715867-511872244-182302923-500\n\tAccount Name:\t\tAdministrator\n\tAccount Domain:\t\tKERB\n\tLogon ID:\t\t0x6327A21\n\nObject:\n\tObject Server:\t\tDS\n\tObject Type:\t\t%{19195a5b-6da0-11d0-afd3-00c04fd930c9}\n\tObject Name:\t\t%{da8c583e-6d51-41d5-8515-782cba5f390b}\n\tHandle ID:\t\t0x0\n\nOperation:\n\tOperation Type:\t\tObject Access\n\tAccesses:\t\tControl Access\n\t\t\t\t\n\tAccess Mask:\t\t0x100\n\tProperties:\t\tControl Access\n\t\t{1131f6ad-9c07-11d1-f79f-00c04fc2dcd2}\n\t{19195a5b-6da0-11d0-afd3-00c04fd930c9}\n\n\nAdditional Information:\n\tParameter 1:\t\t-\n\tParameter 2:\t\t",
    "input": {
      "type": "winlog"
    },
    "@timestamp": "2022-02-09T14:21:43.722Z",
    "ecs": {
      "version": "1.12.0"
    },
    "data_stream": {
      "namespace": "default",
      "type": "logs",
      "dataset": "system.security"
    },
    "host": {
      "hostname": "DCK",
      "os": {
        "build": "14393.693",
        "kernel": "10.0.14393.693 (rs1_release.161220-1747)",
        "name": "Windows Server 2016 Datacenter Evaluation",
        "family": "windows",
        "type": "windows",
        "version": "10.0",
        "platform": "windows"
      },
      "ip": [
        "192.168.15.39",
        "fe80::d4e:3080:ee72:bd1d",
        "10.0.3.15",
        "fe80::5efe:c0a8:f27",
        "fe80::ffff:ffff:fffe",
        "fe80::5efe:a00:30f"
      ],
      "name": "DCK.kerb.com",
      "id": "2d37ee2b-e19c-495f-8540-45bfd43dbb2b",
      "mac": [
        "08:00:27:d3:c9:65"
      ],
      "architecture": "x86_64"
    },
    "event": {
      "agent_id_status": "verified",
      "ingested": "2022-02-09T14:21:33Z",
      "code": "4662",
      "provider": "Microsoft-Windows-Security-Auditing",
      "created": "2022-02-09T14:21:45.914Z",
      "kind": "event",
      "action": "Directory Service Access",
      "dataset": "system.security",
      "outcome": "success"
    }
  },
  "fields": {
    "winlog.event_data.Properties": [
      "%%7688\n\t\t{1131f6ad-9c07-11d1-f79f-00c04fc2dcd2}\n\t{19195a5b-6da0-11d0-afd3-00c04fd930c9}"
    ],
    "elastic_agent.version": [
      "7.16.3"
    ],
    "host.os.name.text": [
      "Windows Server 2016 Datacenter Evaluation"
    ],
    "winlog.provider_guid": [
      "{54849625-5478-4994-A5BA-3E3B0328C30D}"
    ],
    "winlog.provider_name": [
      "Microsoft-Windows-Security-Auditing"
    ],
    "host.hostname": [
      "DCK"
    ],
    "winlog.computer_name": [
      "DCK.kerb.com"
    ],
    "host.mac": [
      "08:00:27:d3:c9:65"
    ],
    "winlog.process.pid": [
      512
    ],
    "winlog.keywords": [
      "Audit Success"
    ],
    "host.os.version": [
      "10.0"
    ],
    "winlog.record_id": [
      "12593"
    ],
    "winlog.logon.id": [
      "0x6327a21"
    ],
    "winlog.event_data.ObjectName": [
      "%{da8c583e-6d51-41d5-8515-782cba5f390b}"
    ],
    "host.os.name": [
      "Windows Server 2016 Datacenter Evaluation"
    ],
    "log.level": [
      "information"
    ],
    "agent.name": [
      "DCK"
    ],
    "host.name": [
      "DCK.kerb.com"
    ],
    "event.agent_id_status": [
      "verified"
    ],
    "winlog.event_data.AccessListDescription": [
      "Control Access"
    ],
    "event.kind": [
      "event"
    ],
    "winlog.activity_id": [
      "{E2864E26-1C39-0000-4F4E-86E2391CD801}"
    ],
    "event.outcome": [
      "success"
    ],
    "winlog.event_data.AccessMask": [
      "0x100"
    ],
    "host.os.type": [
      "windows"
    ],
    "input.type": [
      "winlog"
    ],
    "agent.hostname": [
      "DCK"
    ],
    "data_stream.type": [
      "logs"
    ],
    "host.architecture": [
      "x86_64"
    ],
    "event.provider": [
      "Microsoft-Windows-Security-Auditing"
    ],
    "event.code": [
      "4662"
    ],
    "agent.id": [
      "e9c742aa-7cc9-4ec9-b8ca-fe3e493552a8"
    ],
    "ecs.version": [
      "1.12.0"
    ],
    "event.created": [
      "2022-02-09T14:21:45.914Z"
    ],
    "winlog.event_data.OperationType": [
      "Object Access"
    ],
    "agent.version": [
      "7.16.3"
    ],
    "host.os.family": [
      "windows"
    ],
    "winlog.event_data.SubjectUserSid": [
      "S-1-5-21-2581715867-511872244-182302923-500"
    ],
    "winlog.process.thread.id": [
      13580
    ],
    "host.os.build": [
      "14393.693"
    ],
    "winlog.event_data.AccessList": [
      "%%7688\n\t\t\t\t"
    ],
    "host.ip": [
      "192.168.15.39"
    ],
    "agent.type": [
      "filebeat"
    ],
    "event.module": [
      "system"
    ],
    "winlog.event_data.AdditionalInfo": [
      "-"
    ],
    "winlog.event_data.SubjectLogonId": [
      "0x6327a21"
    ],
    "host.os.kernel": [
      "10.0.14393.693 (rs1_release.161220-1747)"
    ],
    "winlog.api": [
      "wineventlog"
    ],
    "elastic_agent.snapshot": [
      false
    ],
    "host.id": [
      "2d37ee2b-e19c-495f-8540-45bfd43dbb2b"
    ],
    "winlog.task": [
      "Directory Service Access"
    ],
    "elastic_agent.id": [
      "e9c742aa-7cc9-4ec9-b8ca-fe3e493552a8"
    ],
    "winlog.event_data.ObjectType": [
      "%{19195a5b-6da0-11d0-afd3-00c04fd930c9}"
    ],
    "data_stream.namespace": [
      "default"
    ],
    "winlog.event_data.SubjectUserName": [
      "Administrator"
    ],
    "message": [
      "An operation was performed on an object.\n\nSubject :\n\tSecurity ID:\t\tS-1-5-21-2581715867-511872244-182302923-500\n\tAccount Name:\t\tAdministrator\n\tAccount Domain:\t\tKERB\n\tLogon ID:\t\t0x6327A21\n\nObject:\n\tObject Server:\t\tDS\n\tObject Type:\t\t%{19195a5b-6da0-11d0-afd3-00c04fd930c9}\n\tObject Name:\t\t%{da8c583e-6d51-41d5-8515-782cba5f390b}\n\tHandle ID:\t\t0x0\n\nOperation:\n\tOperation Type:\t\tObject Access\n\tAccesses:\t\tControl Access\n\t\t\t\t\n\tAccess Mask:\t\t0x100\n\tProperties:\t\tControl Access\n\t\t{1131f6ad-9c07-11d1-f79f-00c04fc2dcd2}\n\t{19195a5b-6da0-11d0-afd3-00c04fd930c9}\n\n\nAdditional Information:\n\tParameter 1:\t\t-\n\tParameter 2:\t\t"
    ],
    "winlog.event_id": [
      "4662"
    ],
    "event.ingested": [
      "2022-02-09T14:21:33.000Z"
    ],
    "event.action": [
      "Directory Service Access"
    ],
    "@timestamp": [
      "2022-02-09T14:21:43.722Z"
    ],
    "winlog.channel": [
      "Security"
    ],
    "host.os.platform": [
      "windows"
    ],
    "data_stream.dataset": [
      "system.security"
    ],
    "winlog.opcode": [
      "Info"
    ],
    "winlog.event_data.ObjectServer": [
      "DS"
    ],
    "winlog.event_data.HandleId": [
      "0x0"
    ],
    "agent.ephemeral_id": [
      "1101fb45-3ad8-4ec4-9fbc-2e40a4801624"
    ],
    "winlog.event_data.AccessMaskDescription": [
      "Control Access"
    ],
    "winlog.event_data.SubjectDomainName": [
      "KERB"
    ],
    "event.dataset": [
      "system.security"
    ]
  }
}
```

</details>

### CODE REVIEW

**Review by DefSecSentinel — APPROVED** [review]
LGTM

File: etc/non-ecs-schema.json [hunk 1]
```diff
@@ -22,15 +23,16 @@
...
+        "Properties": "keyword",
         "ParentProcessId": "keyword",
         "RelativeTargetName": "keyword",
         "ShareName": "keyword",
         "SubjectLogonId": "keyword",
+        "SubjectUserName": "keyword",
         "TargetImage": "keyword",
         "TargetLogonId": "keyword",
         "TargetProcessGUID": "keyword",
         "TargetSid": "keyword"
```
Comment (brokensound77) [inline etc/non-ecs-schema.json:35]:
```suggestion
        "TargetSid": "keyword",
      }
```

File: etc/non-ecs-schema.json [hunk 2]
```diff
@@ -22,14 +23,16 @@
...
         "ParentProcessId": "keyword",
         "RelativeTargetName": "keyword",
         "ShareName": "keyword",
         "SubjectLogonId": "keyword",
+        "SubjectUserName": "keyword",
         "TargetImage": "keyword",
         "TargetLogonId": "keyword",
         "TargetProcessGUID": "keyword",
-        "TargetSid": "keyword"
+        "TargetSid": "keyword",
```
Comment (brokensound77) [inline etc/non-ecs-schema.json:35]:
```suggestion
        "TargetSid": "keyword"
```

**Review by brokensound77 — APPROVED** [review]
Nice guide and query! LGTM 👍 

I always appreciate that you include the data ++

**Pull request merged**